.github/workflows/dart_ci.yml

```diff
@@ -94,13 +97,23 @@ jobs:
       - name: Print Dart SDK version
         run: dart --version
 
+      - id: link
+        name: Override over_react dependency with local path
+        run: cd ../.. && pub get && dart tool/travis_link_plugin_deps.dart
+
       - id: install
         name: Install dependencies
         run: pub get
+        if: always() && steps.link.outcome == 'success'
 
-      - name: Run analyzer_plugin checks
-        run: |
-          dart tool/travis_link_plugin_deps.dart
-          cd ./tools/analyzer_plugin
-          ./tool/travis.sh
+      - name: Validate dependencies
+        run: pub run dependency_validator --no-fatal-pins -i analyzer,build_runner,build_web_compilers,built_value_generator
```

Comment:
fyi if you want to upgrade to dependency_validator v2 or v3, then it should detect the executable from `build_runner` and the builders from the last two, and then you'd only need to ignore `analyzer` which you could do statically. You'd also have to drop `--no-fatal-pins` and ignore whichever packages are ignored.

Reply:
(probably out-of-scope since this is a small PR, just thought I'd throw it out there for future reference)